src/circuits/joinsplit.tcc

```diff
@@ -21,64 +20,64 @@ template<
 class joinsplit_gadget : libsnark::gadget<FieldT>
 {
 private:
-    // Multipacking gadgets for the inputs (nullifierS, commitmentS, val_pub_in,
-    // val_pub_out) (the root is a field element) `NumInputs + NumOutputs`
-    // because we pack the nullifiers (Inputs of JS = NumInputs), the
-    // commitments (Output of JS = NumOutputs) AND the v_pub_out taken out of
-    // the mix (+1) AND the public value v_pub_in that is put into the mix (+1)
-    // AND the signature hash h_sig (+1) AND the malleability tags h_iS (+
-    // NumInputs)
+    /*
+     * Multipacking gadgets for the inputs (nullifierS, commitmentS, val_pub_in,
+     *val_pub_out) (the root is a field element) `NumInputs + NumOutputs`
+     *because we pack the nullifiers (Inputs of JS = NumInputs), the commitments
+     *(Output of JS = NumOutputs) AND the v_pub_out taken out of the mix (+1)
+     *AND the public value v_pub_in that is put into the mix (+1) AND the
+     *signature hash h_sig (+1) AND the malleability tags h_iS (+ NumInputs)
+     **/
+    size_t bits = 2 * 64 + 3 * (1 + 2 * NumInputs + NumOutputs);
+    size_t log_r =
+        253; // log_2(21888242871839275222246405745257275088548364400416034343698204186575808495617);
     std::array<
         libsnark::pb_variable_array<FieldT>,
-        NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs>
+        NumInputs + NumOutputs + 1 + NumInputs + 1>
         packed_inputs;
     std::array<
         libsnark::pb_variable_array<FieldT>,
-        NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs>
+        NumInputs + NumOutputs + 1 + NumInputs + 1>
         unpacked_inputs;
 
-    // We use an array of multipackers here instead of a single packer that
-    // packs everything. This leads to more public inputs (and thus affects a
-    // little bit the verification time) but this makes easier to retrieve the
-    // root and each nullifiers from the public inputs
+    /* We use an array of multipackers here instead of a single packer that
+     *packs everything. This leads to more public inputs (and thus affects a
+     *little bit the verification time) but this makes easier to retrieve the
+     *root and each nullifiers from the public inputs
+     **/
     std::array<
         std::shared_ptr<libsnark::multipacking_gadget<FieldT>>,
-        NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs>
+        NumInputs + NumOutputs + 1 + NumInputs + 1>
         packers;
 
+    // TODO: Remove ZERO and pass it in the constructor
```

Comment:
This is not on develop, I guess this branch has forked some time ago from develop. Please make sure to apply your changes on top of the latest changes on develop to avoid re-introducing obsolete changes